src/staged_discrete.jl

```diff
@@ -0,0 +1,180 @@
+
+# Constructor
+    StagedDiscreteBelief(pomdp, b::Vector{Float64}, st::Int; check::Bool=true)
+
+# Fields
+- `pomdp` : the POMDP problem wrapped in FixedHorizon(PO)MDPWrapper struct
+            from FiniteHorizonPOMDPs.jl
+- `state_list` : a vector of ordered states in a given stage
+                (Tuple of Infinite Horizon states and stage number)
+                eg. (:StateX, 1)
+- `b` : the probability vector
+- `st` : number of POMDP's stage that `StagedDiscreteBelief` represents
+"""
+struct StagedDiscreteBelief{P<:FiniteHorizonPOMDPs.FHWrapper, S}
+    pomdp::P
+    state_list::Vector{S}
+    b::Vector{Float64}
+    st::Int
+end
+
+
+function StagedDiscreteBelief(pomdp::FiniteHorizonPOMDPs.FHWrapper, b::Vector{Float64}, st; check::Bool=true)
+    if check
+        if !isapprox(sum(b), 1.0, atol=0.001)
+            @warn """
+                  b in StagedDiscreteBelief(pomdp, b) does not sum to 1.
+
+                  To suppress this warning use `StagedDiscreteBelief(pomdp, b, check=false)`
+                  """
+        end
+        if !all(0.0 <= p <= 1.0 for p in b)
+            @warn """
+                  b in StagedDiscreteBelief(pomdp, b) contains entries outside [0,1].
+
+                  To suppress this warning use `StagedDiscreteBelief(pomdp, b, check=false)`
+                  """
+        end
+    end
+    return StagedDiscreteBelief(pomdp, ordered_stage_states(pomdp, st), b, st)
+end
+
+
+function StagedDiscreteBelief(pomdp::FiniteHorizonPOMDPs.FHWrapper,
+        d::FiniteHorizonPOMDPs.InStageDistribution; check::Bool=true)
+    st = FiniteHorizonPOMDPs.stage(d)
+    state_list = ordered_stage_states(pomdp, st)
+    b = [pdf(d, ss) for ss in state_list]
+    if check
+        if !isapprox(sum(b), 1.0, atol=0.001)
+            @warn """
+                  b in StagedDiscreteBelief(pomdp, b) does not sum to 1.
+
+                  To suppress this warning use `StagedDiscreteBelief(pomdp, b, check=false)`
+                  """
+        end
+        if !all(0.0 <= p <= 1.0 for p in b)
+            @warn """
+                  b in StagedDiscreteBelief(pomdp, b) contains entries outside [0,1].
+
+                  To suppress this warning use `StagedDiscreteBelief(pomdp, b, check=false)`
+                  """
+        end
+    end
+    return StagedDiscreteBelief(pomdp, state_list, b, st)
+end
+
+stage(b::StagedDiscreteBelief) = b.st
+
+"""
+     uniform_staged_belief(pomdp::FiniteHorizonPOMDPs.FHWrapper)
+
+Return a StagedDiscreteBelief with equal probability for each state in a given stage.
+"""
+function uniform_staged_belief(pomdp::FiniteHorizonPOMDPs.FHWrapper; st::Int=1)
+    state_list = ordered_stage_states(pomdp, st)
+    ns = length(state_list)
+    return StagedDiscreteBelief(pomdp, state_list, ones(ns) / ns, st)
+end
+
+function uniform_belief(pomdp::FiniteHorizonPOMDPs.FHWrapper)
+    return uniform_staged_belief(pomdp; st=1)
+end
+
+
+function POMDPs.pdf(b::StagedDiscreteBelief, ss::Tuple{<:Any, Int})
```

Comment:
```suggestion
function POMDPs.pdf(b::StagedDiscreteBelief, ss)
```
The state in for a finite horizon POMDP might not be a tuple. Tuples are only used in the special case of `FHWrapper`